std/io.zig

```diff
@@ -1067,6 +1067,51 @@ test "io.readLineFrom" {
     testing.expectEqualSlices(u8, "Line 1Line 22Line 333", buf.toSlice());
 }
 
+pub fn readLineIterator(stream: var, buf: *std.Buffer) type {
+    const StreamType = @typeOf(stream);
+    const Iterator = struct {
+        stream: StreamType,
+        buf: *std.Buffer,
+        pub fn next(self: *@This()) !?[]u8 {
+            return readLineFrom(self.stream, self.buf) catch |err| {
+                return switch (err) {
+                    error.EndOfStream => null,
```

Comment:
shouldn't this just return the final (unterminated) line?

Reply:
In this case the stream has ended so there's no line to return.

Reply:
There is though: at least from my reading of the code, there's data left in `buf`.

Reply:
There shouldn't be. Plus the test would fail if that was the case.

Reply:
Unless, could readLineFrom return EndOfStream with data?

Reply:
Looking through readLineFrom, it looks like you're right. Will have to push a fix.